### PR TITLE
HIVE-29342: Iceberg: [V3] Display initial and write defaults in descr…

### PR DESCRIPTION
**Pull request by tanishq-chugh**
…ibe formatted column output

### What changes were proposed in this pull request?

Add Initial Default and Write Default values in DESCRIBE FORMATTED column output for iceberg table columns

### Why are the changes needed?

Improvement in DESCRIBE FORMATTED column output for iceberg table columns

### Does this PR introduce _any_ user-facing change?

Yes

### How was this patch tested?

Manual Testing and Qtest

### CODE REVIEW

File: iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out [hunk 1]
```diff
@@ -0,0 +1,406 @@
...
+Compressed:         	No                  	 
+Sort Columns:       	[]                  	 
+	 	 
+# Constraints	 	 
+	 	 
+# Default Constraints	 	 
+Table:              	default.ice_parq    	 
+Constraint Name:    	#### A masked pattern was here ####	 
+Column Name:age     	Default Value:25    	 
+
```
Comment (ayushtkn) [inline iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out:113]: Initial Default Should come here in Default Constraints block, not in schema

File: iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+	 	 
+# Constraints	 	 
+	 	 
+# Default Constraints	 	 
+Table:              	default.ice_parq    	 
+Constraint Name:    	#### A masked pattern was here ####	 
+Column Name:age     	Initial Default Value:25	Write Default Value:25
+	 	 
+Constraint Name:    	#### A masked pattern was here ####	 
+Column Name:point   	Initial Default Value:	Write Default Value:'x:100,y:99'
```
Comment (ayushtkn) [inline iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out]: Initial Default should show up when there is a value, not otherwise

File: iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+OutputFormat:       	org.apache.iceberg.mr.hive.HiveIcebergOutputFormat	 
+Compressed:         	No                  	 
+Sort Columns:       	[]                  	 
+	 	 
+# Constraints	 	 
+	 	 
+# Default Constraints	 	 
+Table:              	default.ice_orc     	 
+Constraint Name:    	#### A masked pattern was here ####	 
+Column Name:point   	Initial Default Value:	Write Default Value:'x:100,y:99'
```
Comment (ayushtkn) [inline iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out]: While adding the default string is ``DEFAULT '{"x":100,"y":99}'``, it should be exactly same, can you change it use the same thing

**Review by ayushtkn — COMMENTED** [review]
Thanx @tanishq-chugh 

Can you add a test with some nested struct default & case where the write default is set to `null` but the initial default is there

File: iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out (hunk 2 above)
Comment (tanishq-chugh) [reply iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out]: Updated to ensure initial default / write default appears only when they have a value in commit: [69ba9ee](https://github.com/apache/hive/pull/6367/commits/69ba9eefb6b6cf5b76f4c1f9ed1576b8cc2f4030)

File: iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out (hunk 3 above)
Comment (tanishq-chugh) [reply iceberg/iceberg-handler/src/test/results/positive/iceberg_defaults_in_desc_table.q.out]: Addressed this in commit - [69ba9ee](https://github.com/apache/hive/pull/6367/commits/69ba9eefb6b6cf5b76f4c1f9ed1576b8cc2f4030)

**Comment by tanishq-chugh** [comment]
Thanks for checking this @ayushtkn !
I have added queries (for test cases) in the same Qtest for the cases with nested struct default (great catch, this was broken, have fixed in commit -[5a0fcac](https://github.com/apache/hive/pull/6367/commits/5a0fcac25f9d0670c019036c4ee7df03aecef6dc) ) & initial default is set but write default is null.
